database_api.py

```
@@ -313,6 +316,184 @@
+    password = request.json.get('password')
+    
+    # LDAP injection vulnerability
+    ldap_filter = f"(&(uid={username})(userPassword={password}))"
+    
+    # Simulated LDAP query (vulnerable to injection)
+    return jsonify({"authenticated": True, "filter": ldap_filter})
+
+@app.route('/api/redis_cache', methods=['POST'])
+def redis_operation():
+    """Redis operations - Command injection"""
+    import redis
```

Ruff (0.12.2)

Redefinition of unused `redis` from line 22

Remove definition: `redis`

(F811)